File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py

def solution(numbers, target):
    answer = 0
    i = 0
    answer1 = sum_number(numbers, target - numbers[i], i, len(numbers), answer)
    answer2 = sum_number(numbers, target + numbers[i], i, len(numbers), answer)
    # print(answer1 , answer2)
    return answer1 + answer2

def sum_number (numbers, target, i, end, answer) :
    
    # print(numbers[i], i, target,answer)
    if i == end - 1:
        # print("end")
        if target == 0 : 
            answer += 1
            # print(answer)
        return answer
        
    answer = sum_number(numbers, target - numbers[i+1], i+1, end, answer)
    answer = sum_number(numbers, target + numbers[i+1], i+1, end, answer)
    return answer 

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.java
class Solution {
    public int solution(int[] numbers, int target) {
        int answer = 0;
        answer = rop(numbers,0,target, answer);
        return answer;
    }
    
    public int rop(int[] numbers, int index, int target, int answer){
        
        if(index >= numbers.length){
            if(target == 0){
                answer += 1; 
            }
            return answer; 
            
        }
        
        answer = rop(numbers , index+1 , target-numbers[index], answer);
        answer = rop(numbers, index+1, target+numbers[index],answer);
        
        return answer; 
        
    }
}